nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/http/ErrorHandlers.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.webserver.http;
+
+import java.io.UncheckedIOException;
+
+import io.helidon.common.http.BadRequestException;
+import io.helidon.common.http.DirectHandler;
+import io.helidon.common.http.HttpException;
+import io.helidon.common.http.InternalServerException;
+import io.helidon.common.http.RequestException;
+import io.helidon.nima.webserver.CloseConnectionException;
+import io.helidon.nima.webserver.ConnectionContext;
+
+/**
+ * Http routing Error handlers.
+ */
+public class ErrorHandlers {
+    ErrorHandlers() {
+    }
+
+    /**
+     * Run a task and handle the errors, if any. Uses error handlers configured on routing.
+     * Correctly handles all expected (and unexpected) exceptions that can happen in filters and routes.
+     *
+     * @param ctx connection context
+     * @param request HTTP server request
+     * @param response HTTP server response
+     * @param task task to execute
+     */
+    public void runWithErrorHandling(ConnectionContext ctx, ServerRequest request, ServerResponse response, Runnable task) {
+        try {
+            task.run();
+        } catch (CloseConnectionException | UncheckedIOException e) {
+            // these errors must "bubble up"
+            throw e;
+        } catch (RequestException e) {
+            handleRequestException(ctx, response, e);
+        } catch (BadRequestException e) {
+            // bad request exception MUST be handled by direct handlers
+            handleRequestException(ctx, response, RequestException.builder()
+                    .message(e.getMessage())
+                    .cause(e)
+                    .type(DirectHandler.EventType.BAD_REQUEST)
+                    .status(e.status())
+                    .setKeepAlive(e.keepAlive())
+                    .build());
+        } catch (InternalServerException e) {
+            // this is the place error handling must be done
+            // check if error handler exists for cause - if so, use it
+            if (hasErrorHandler(e.getCause())) {
+                handleError(ctx, request, response, e.getCause());
+            } else {
+                handleError(ctx, request, response, e);
+            }
+        } catch (HttpException e) {
+            handleError(ctx, request, response, e);
+        } catch (RuntimeException e) {
+            handleError(ctx, request, response, e);
+        }
```

Comment:
I would rather catch Throwable to be sure

Reply:
I don't think we want to handle "Throwable" - that would contain things like `OutOfMemoryError` and that should not be a case for exception handling by users. We can add it once we add error handling to Nima...